discord_connector.py

```diff
@@ -0,0 +1,436 @@
+
+    def _handle_fetch_message(self, param):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        channel_id = param['channel_id']
+        message_id = param['message_id']
+
+        status, message = self.fetch_message(channel_id, message_id, action_result)
+        if not status:
+            return action_result.set_status(phantom.APP_ERROR,
+                                            "action result: fetching message {} ended with failure".format(message_id))
+
+        attachments, embeds = self.create_artifacts(message)
+        message = self.parse_message(message, attachments, embeds)
+
+        action_result.add_data(message)
+        summary = action_result.update_summary({})
+        summary['action result: '] = "fetching message {} ended with success".format(message_id)
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def fetch_message(self, channel_id, message_id, action_result) -> discord.Message or None:
+        status, channel = self.run_in_loop(self._guild.fetch_channel(channel_id), action_result,
+                                           error_message=DISCORD_ERROR_FETCHING_CHANNEL)
+        if not status:
+            return status, None
+        status, message = self.run_in_loop(channel.fetch_message(message_id), action_result,
+                                           error_message=DISCORD_ERROR_FETCHING_MESSAGE)
+
+        return status, message
+
+    def create_artifacts(self, message):
+        container_id = self.get_container_id()
+        attachments = []
+        embeds = []
+
+        if message.embeds:
+            self.save_progress("working on embeds")
+            for embed in message.embeds:
+                embeds.append(self.create_embed_artifact(embed, container_id))
+
+        if message.attachments:
+            self.save_progress("working on attachments")
+            for attachment in message.attachments:
+                attachments.append(self.create_attachment_artifact(attachment, container_id))
+
+        return attachments, embeds
+
+    def create_embed_artifact(self, embed, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"embed: {embed.title}",
+            cef={"URL": embed.url, "Description": embed.description}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+    def create_attachment_artifact(self, attachment, container_id):
+        artifact = Artifact(
+            container_id=container_id,
+            name=f"attachment: {attachment.filename}",
+            cef={"URL": attachment.url, "Description": attachment.description, "Type": attachment.content_type}
+        )
+        return self.save_artifact_to_soar(dataclasses.asdict(artifact))
+
+    def save_artifact_to_soar(self, artifact):
```

Comment:
Wouldn't it be better to accept artifact as instance of `Artifact` instead of dict? This is pretty confusing. See :147